src/error.rs

```diff
@@ -53,6 +54,8 @@ impl<'tcx> Error for EvalError<'tcx> {
         match *self {
             EvalError::FunctionPointerTyMismatch(..) =>
                 "tried to call a function through a function pointer of a different type",
+            EvalError::ZstPointerDeref =>
+                "tried to deref a pointer to the ZST allocation",
```

Comment:
No. A dereference *should always succeed*. What it does is take a pointer and treat it as a memory location.
*All possible pointer-sized values* are valid memory locations, if you can read the value (i.e. not undefined).

Now what could error is *accesses on those locations*. A 0-byte access does not exist, by definition.
If you have checks on a method that may be used for 0-length byte transfers, *return early for `length == 0`*.